caterva2/api_utils.py

```diff
@@ -62,20 +62,28 @@ def parse_slice(string):
     return tuple(obj)
 
 
-def get_download_url(path, host, params):
-    download_ = params.get('download', False)
-    if download_ and params.get('slice_') is not None:
-        raise ValueError('Cannot download a slice')
-    response = httpx.get(f'http://{host}/api/download/{path}', params=params)
+def fetch_data(path, host, params):
+    if 'as_schunk' not in params:
+        params['as_schunk'] = blosc2_is_here
+    response = httpx.get(f'http://{host}/api/fetch/{path}', params=params)
     response.raise_for_status()
-
-    if download_:
-        return response.json()
-
     data = response.content
-    # TODO: decompression is not working yet. HTTPX does this automatically?
-    # data = zlib.decompress(data)
-    return pickle.loads(data)
+    # Try different deserialization methods
```

Comment:
All the more reason to use `prefer_schunk` rather than `as_schunk` (see the other comment).

Reply:
(Again, it would be nice if the response included a proper `Content-Type:`, but we'd need to invent them. But this also applies to other responses, so it can be delayed.)